plugins/modules/remote_management/ipmi/ipmi_boot.py

```diff
@@ -162,10 +169,15 @@ def main():
     if state == 'absent' and bootdev == 'default':
         module.fail_json(msg="The bootdev 'default' cannot be used with state 'absent'.")
 
+    if module.params['key']:
+        key = bytes.fromhex(module.params['key'])
```

Comment:
On Python 2:
```
>>> bytes.fromhex('11')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'str' has no attribute 'fromhex'
```
You probably want to use `binascii.unhexlify`.
```
>>> import binascii
>>> binascii.unhexlify(b'001122ff')
b'\x00\x11"\xff'
```

Reply:
(https://docs.python.org/2/library/binascii.html#binascii.unhexlify)

Reply:
Thank you for the feedback, it is much appreciated!
I will check more about this & update in the coming days. Hopefully there could be other feedback as well.

Reply:
Should there be input validation - i.e. if the call to unhexlify fails when e.g. user passed an ASCII string there should be a nice error message?
Or maybe if the unhexlify fails it should attempt to pass the key directly as `kg` without translation?

At least on Dell DRAC BMCs the key is "always" (tested DRAC5 until DRAC8) configured in this hex format but I do not know about other platforms.

Reply:
> Should there be input validation - i.e. if the call to unhexlify fails when e.g. user passed an ASCII string there should be a nice error message?

Definitely yes. You should catch the exception and call `module.fail_json()`. Even if you just pass on the exception's value with some text prepended (that the user knows which option the error refers to).

>  Or maybe if the unhexlify fails it should attempt to pass the key directly as `kg` without translation?

I would not do that. It makes errors usually harder to debug, since you need to know the exact input data to be able to tell what went wrong. (And since the input data here is secret, this would be a support nightmare :) )

Reply:
> > Should there be input validation - i.e. if the call to unhexlify fails when e.g. user passed an ASCII string there should be a nice error message?
> 
> Definitely yes. You should catch the exception and call `module.fail_json()`. Even if you just pass on the exception's value with some text prepended (that the user knows which option the error refers to).

Just a reminder that the call is already wrapped by a `try/except` block. So, unless you want to have a specific error-handling for this case, that base is already covered.

Reply:
@russoz it is not part of a try/except. The try only starts a couple of lines below this.

Reply:
Oops, you're sright, sorry about that. Anyway, it should be easy to drag the try a couple of lines up and include this part .

Reply:
I would still create a new try/except just for this, so that the message is actually helpful to the user (by mentioning which option this failure is about).

Reply:
Changed the hex string handling to the suggested python 2 compatible way and added exception with an explanation error message.